Mathlib/Analysis/Analytic/Constructions.lean

```diff
@@ -1233,3 +1233,106 @@ theorem HasFPowerSeriesWithinAt.unshift (hf : HasFPowerSeriesWithinAt f pf s x)
   hrf.unshift.hasFPowerSeriesWithinAt
 
 end
+
+/-!
+### Composition with a linear map
+-/
+
+section compContinuousLinearMap
+
+variable {u : E →L[𝕜] F} {f : F → G} {pf : FormalMultilinearSeries 𝕜 F G} {s : Set F} {x : E}
+  {r : ℝ≥0∞}
+
+theorem HasFPowerSeriesWithinOnBall.compContinuousLinearMap
+    (hf : HasFPowerSeriesWithinOnBall f pf s (u x) r) :
+    HasFPowerSeriesWithinOnBall (f ∘ u) (pf.compContinuousLinearMap u) (u ⁻¹' s) x (r / ‖u‖₊) where
+  r_le := by
+    calc
+      _ ≤ pf.radius / ↑‖u‖₊ := by
+        gcongr
+        exact hf.r_le
+      _ ≤ _ := pf.radius_compContinuousLinearMap_ge u
+  r_pos := by
+    simp only [ENNReal.div_pos_iff, ne_eq, coe_ne_top, not_false_eq_true, and_true]
+    exact pos_iff_ne_zero.mp hf.r_pos
+  hasSum := by
+    intro y hy1 hy2
+    convert hf.hasSum (y := u y) _ _
+    · simp
+    · simp at hy1 ⊢
+      rcases hy1 with (hy1 | hy1) <;> simp [hy1]
+    by_cases hu_zero : ‖u‖₊ = 0
+    · simp only [nnnorm_eq_zero] at hu_zero
+      simp [hu_zero, hf.r_pos]
+    cases r with
+    | top => simp
+    | coe r =>
+      rw [← ENNReal.coe_div hu_zero] at hy2
+      simp only [Metric.emetric_ball_nnreal, NNReal.coe_div, coe_nnnorm, Metric.mem_ball,
+        dist_zero_right] at hy2 ⊢
+      calc
+        _ ≤ _ := u.le_opNorm y
+        _ < _ := by
+          rwa [lt_div_iff₀'] at hy2
+          simpa using hu_zero
+
+theorem HasFPowerSeriesOnBall.compContinuousLinearMap (hf : HasFPowerSeriesOnBall f pf (u x) r) :
+    HasFPowerSeriesOnBall (f ∘ u) (pf.compContinuousLinearMap u) x (r / ‖u‖₊) where
```

Comment:
This version should follow almost immediately from the previous lemma and [hasFPowerSeriesWithinOnBall_univ](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/Analytic/Basic.html#hasFPowerSeriesWithinOnBall_univ)